flatfs/flatfs.go

```diff
@@ -86,6 +82,17 @@ func (fs *Datastore) makePrefixDir(dir string) error {
 	return nil
 }
 
+func (fs *Datastore) makePrefixDirNoSync(dir string) error {
+	if err := os.Mkdir(dir, 0777); err != nil {
+		// EEXIST is safe to ignore here, that just means the prefix
+		// directory already existed.
```

Comment:
Whenever you see me not using MkdirAll, it's 99% safe to assume it's intentional. The parent dir should exist by the time that code runs; if not, I actually want to see an error. This also makes typos near the beginning of the path less likely to create weird shadow hierarchies. The Init function that creates the parent is https://github.com/ipfs/go-ipfs/blob/b1bb16f0f291d66b5902b8a6cefc2eac204a31e2/repo/fsrepo/fsrepo.go#L263